### PR TITLE
Merge group files endpoints

### PR DESCRIPTION
**Pull request by GGP1**
|Related issue|
|---|
| Closes https://github.com/wazuh/wazuh/issues/20851 |

## Description

Merges the `/groups/:group_id/files/:file_name/json` and `/groups/:group_id/files/:file_name/xml` endpoints into a single one that takes the `raw` parameter which defines whether the response content will be formatted in json or plain text. The response content-type is defined based on the file extension retrieved.

## Logs/Alerts example

<details><summary>Get merged.mg in plain text</summary>

`GET /groups/default/files/merged.mg?raw=true`

```
#default
!228 ar.conf
restart-ossec0 - restart-ossec.sh - 0
restart-ossec0 - restart-ossec.cmd - 0
restart-wazuh0 - restart-ossec.sh - 0
restart-wazuh0 - restart-ossec.cmd - 0
restart-wazuh0 - restart-wazuh - 0
restart-wazuh0 - restart-wazuh.exe - 0
!76 agent.conf
<agent_config>

  <!-- Shared agent configuration here -->

</agent_config>
!28411 cis_apache2224_rcl.txt
...
```

</details>

<details><summary>Get merged.mg in JSON</summary>

`GET /groups/default/files/merged.mg`

```json
{
    "data": [
        {
            "file_name": "ar.conf",
            "file_size": 228,
            "file_content": "restart-ossec0 - restart-ossec.sh - 0\nrestart-ossec0 - restart-ossec.cmd - 0\nrestart-wazuh0 - restart-ossec.sh - 0\nrestart-wazuh0 - restart-ossec.cmd - 0\nrestart-wazuh0 - restart-wazuh - 0\nrestart-wazuh0 - restart-wazuh.exe - 0\n"
        },
        {
            "file_name": "agent.conf",
            "file_size": 76,
            "file_content": "<agent_config>\n\n  <!-- Shared agent configuration here -->\n\n</agent_config>\n"
        },
        ...
 	]
}
```

</details>

<details><summary>Get file content using a configuration type</summary>

`GET /groups/default/files/agent.conf?type=conf`

```json
{
    "data": {
        "total_affected_items": 1,
        "affected_items": [
            {
                "filters": {},
                "config": {}
            }
        ]
    },
    "error": 0
}
```

</details>

> Failed checks are not related to the changes introduced.

### CODE REVIEW

**Review by EduLeon12 — APPROVED** [review]
Great work @GGP1

 LGTM!,

My only question is if it would be better for readibility purposes to change the raw default value to a bool `False` instead of `None.

**Comment by GGP1** [comment]
> My only question is if it would be better for readibility purposes to change the raw default value to a bool `False` instead of `None.

I've seen both in the code, I don't think there's a standard but I agree that using `False` is slightly better so I modified it. Thanks!

File: framework/wazuh/core/configuration.py [hunk 1]
```diff
@@ -770,33 +820,33 @@ def get_agent_conf_multigroup(multigroup_id: str = None, offset: int = 0, limit:
...
         Group was not found.
     WazuhError(1006)
-        agent.conf does not exist or there is a problem with the permissions.
+        The file does not exist or there is a problem with the permissions.
     WazuhError(1104)
         Invalid file type.
 
     Returns
     -------
     dict
```
Comment (Selutario) [inline framework/wazuh/core/configuration.py:848]:
```suggestion
    dict or str
```

**Review by Selutario — CHANGES_REQUESTED** [review]
Looks good! Just some minor changes were requested. Also, I found a strange behaviour. I guess this is expected, but I would like to confirm.

I tried creating a `test.json` file. When the content-type returned is `application/json`, it is seen this way:

> `curl -k -X GET "https://localhost:55000/groups/default/files/test.json?raw=true" -H  "Authorization: Bearer $TOKEN" -v`

image

However, this does not happen when xml content is returned as `application/xml`:
image

Nor when the json file is returned as plain :thinking::
image

**Comment by GGP1** [comment]
> Looks good! Just some minor changes were requested. Also, I found a strange behaviour. I guess this is expected, but I would like to confirm.
> 
> I tried creating a `test.json` file. When the content-type returned is `application/json`, it is seen this way:
> 
> > `curl -k -X GET "https://localhost:55000/groups/default/files/test.json?raw=true" -H  "Authorization: Bearer $TOKEN" -v`
> 
> image

This is expected, the content of the file is returned as a string, and if the Content-Type is `json` then it will be interpreted as a json string instead of an object. This is why new lines are present and the double quotes are escaped. The other content types do not behave the same because they don't need to escape or wrap anything (and thus the output makes it seem like the actual file content).

**Comment by GGP1** [comment]
> Looks good! Just some minor changes were requested. Also, I found a strange behaviour. I guess this is expected, but I would like to confirm.
> 
> I tried creating a `test.json` file. When the content-type returned is `application/json`, it is seen this way:
> 
> > `curl -k -X GET "https://localhost:55000/groups/default/files/test.json?raw=true" -H  "Authorization: Bearer $TOKEN" -v`
> 
> image

@Selutario This is expected, the content of the file is returned as a string, and if the Content-Type is `json` then it will be interpreted as a json string instead of an object.

In order to include all the content inside the string, new lines have to be literal and double quotes have to be escaped. The other content types do not behave the same because they don't need to escape or wrap anything (and thus the output is the actual file content).

**Review by Selutario — APPROVED** [review]
LGTM. I think we'll need to update the documentation too, at least the [RBAC reference](https://documentation.wazuh.com/current/user-manual/api/rbac/reference.html). 

Please, open a PR for this @GGP1

**Pull request merged**